src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/coding/FallThroughCheck.xml

```diff
@@ -15,9 +15,9 @@
  "fallthru", "fall thru", "fall-thru",
  "fallthrough", "fall through", "fall-through"
  "fallsthrough", "falls through", "falls-through" (case-sensitive).
- The comment containing these words must be all on one line,
- and must be on the last non-empty line before the {@code case} triggering
- the warning or on the same line before the {@code case}(ugly, but possible).
+ The comment containing these words must be on the last non-empty line
+ before the {@code case} triggering the warning or on the same line before
+ the {@code case}(ugly, but possible).
```

Comment:
Remove doc as required

Reply:
> > Update in doc to remove:

> The comment containing these words must be all on one line

I don't remember this conversation regarding this but does this mean if I have a suppression word as "suppress me", it will match things like "suppress\nme"? Do you have a test example showing this?

Reply:
Your updated documentation says:
>  ...on the same line before the ...

How is that different then `...must be all on one line...`?

Reply:
> > > Update in doc to remove:
> 
> > The comment containing these words must be all on one line
> 
> I don't remember this conversation regarding this but does this mean if I have a suppression word as "suppress me", it will match things like "suppress \n me"? Do you have a test example showing this?

No, I don't think "suppress \n me" will match.
This is matching by the pattern
```java
```

Reply:
> Your updated documentation says:
> 
> > ...on the same line before the ...
> 
> How is that different then `...must be all on one line...`?

I didn't see the conversion about this part either, but I think "must be all on one line" is redundant and confusing. I don't really understand what it means. The subject of this sentence is "the comment containing these words". However, we only need one "comment" containing these words to suppress this check for that branch.

If we remove this "must be all on one line.", and updated it to
>The comment containing these words must be on the last non-empty line
 before the {@code case} triggering the warning or on the same line before
 the {@code case}(ugly, but possible).

I think this might be able to explain everything?

Reply:
It looks like comment in issue came from https://github.com/checkstyle/checkstyle/pull/12966#issuecomment-1675649800 . This was about "the last comment on that line". This is why the issue has `/* block */ /* fallthru */ // comment`.

If you agree, please restore original wording and add some wording that it can be any comment on that line.

Can you point to keywords where the suppression text is not the last comment on the line. If we don't have one, then create it and point to it.

Example:
/* trigger */ /* comment */ // last
/* comment */ /* fallthru */ // last

Reply:
Let's continue here https://github.com/checkstyle/checkstyle/pull/15405/files?diff=unified&w=0#r1697058346